tests/ui/cast_nan_to_int.stderr

```diff
@@ -0,0 +1,40 @@
+error: casting a known NaN to usize
+  --> $DIR/cast_nan_to_int.rs:5:13
+   |
+LL |     let _ = (0.0_f32 / -0.0) as usize;
+   |             ^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::cast-nan-to-int` implied by `-D warnings`
```

Comment:
Would be worth adding a note explaining that the result will be 0

Reply:
Maybe we should just lint all operations that result in `NaN`? Whether cast or not

Reply:
That would mean we'd need to check far more expressions. This way we only look into the constant evaluation when we have a cast from float to int, which is sufficiently rare (also in earlier versions of Rust, this cast was actually UB, but that's quite a while ago now). Also the risk of false positives seems higher.

I'm fine with retiring this lint once we have one that checks for all NaNs and works reasonably well.